src/uv-common.c

```diff
@@ -155,6 +155,22 @@ static const char* uv__unknown_err_code(int err) {
   return copy != NULL ? copy : "Unknown system error";
 }
 
+#define UV_ERR_NAME_GEN_R(name, _) case UV_ ## name: { strncpy(buf, #name, buflen); break; }
```

Comment:
Can you use `snprintf()` here too?  `strncpy()` clears the remainder of the buffer, which callers may not expect.